src/sagemaker_containers/functions.py

```diff
@@ -66,3 +66,22 @@ def getargspec(fn):  # type: (function) -> inspect.ArgSpec
     elif six.PY3:
         full_arg_spec = inspect.getfullargspec(fn)
         return inspect.ArgSpec(full_arg_spec.args, full_arg_spec.varargs, full_arg_spec.varkw, full_arg_spec.defaults)
+
+
+def error_wrapper(fn, error_class):  # type: (function or None, Exception) -> ...
+    """Wraps function fn in a try catch block that re-raises error_class. Returns None if fn is None.
+
+    Args:
+        fn (function): function to wrapped
+        error_class (Exception): Error class to be re-raised
+
+    Returns:
+        (object): fn wrapped in a try catch.
+    """
+    def wrapper(*args, **kwargs):
+        try:
+            return fn(*args, **kwargs)
+        except Exception as e:
+            six.raise_from(error_class(e), e)
+
+    return fn and wrapper
```

Comment:
I wouldn't have this branch to return None if None is passed in, since it still doesn't validate correctly if any other type is passed in. If you want you could validate if the passed in fn is callable instead. See note below in Transformer init as well, since this affects that.

Reply:
Yeah... I was hoping we could find a more elegant alternative than this. I feel that we should not even check if it None or not here... any thoughts?

Reply:
Yea, I think either we should do full validation (check if its callable), or no validation. We shouldn't special case None. 

If we change the init to use the slightly more verbose " ... if ... else ... " syntax then we don't need the None case.

Reply:
Got it, in the comment below. Let's keep it simple.